pkg/ast/spl/spl.peg

```diff
@@ -505,6 +442,170 @@ Start <- SPACE? initialSearch:(InitialSearchBlock) filterBlocks:(FilterBlock)* q
 
     return q, nil
 }
+/ SPACE? PIPE CMD_GENTIMES SPACE genTimesOption:(GenTimesOptionList) queryAggBlocks:(QueryAggergatorBlock)* SPACE? EOF {
+    var q ast.QueryStruct
+    q.PipeCommands = &structs.QueryAggregators{
+        PipeCommandType: structs.GenerateEventType,
+        GenerateEvent: &structs.GenerateEvent{
+            GenTimes: genTimesOption.(*structs.GenTimes),
+        },
+    }
+
+    if queryAggBlocks != nil {
+        queryAggSlice := queryAggBlocks.([]any)
+
+        if len(queryAggSlice) > 0 {
+            curQueryAgg := q.PipeCommands
+            // Link the remaining chains.
+            for i := range queryAggSlice {
+                queryAgg := queryAggSlice[i].(*structs.QueryAggregators)
+                curQueryAgg.Next = queryAgg
+
+                // Go to the end of this chain.
+                for ; curQueryAgg.Next != nil; curQueryAgg = curQueryAgg.Next {}
+            }
```

Comment:
May be we can extract this as a separate function? So that we don't do duplication?

Reply:
addressed in the latest commit.